bands/site_sections/bands.py

```diff
@@ -9,143 +9,172 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, band_id, id=None, message='', **params):
+        band = session.band(band_id)
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
                 message = 'You must tell us how many people are in your band'
-            elif not band.poc_phone:
+            elif not band_info.poc_phone:
                 message = 'You must enter an on-site point-of-contact cellphone number'
-            elif not band.arrival_time:
+            elif not band_info.arrival_time:
                 message = 'You must enter your expected arrival time'
-            elif band.bringing_vehicle and not band.vehicle_info:
+            elif band_info.bringing_vehicle and not band_info.vehicle_info:
                 message = 'You must provide your vehicle information'
             else:
+                band.info = band_info
+                session.merge(band_info)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'Your band information has been uploaded')
 
         return {
             'band': band,
+            'band_info': band.info or band_info,
```

Comment:
This was done this way to prevent an error where the form would populate with a new band_info ID every time you loaded the page, regardless of whether band.info existed. Passing the DB object still lets you keep form edits even when your form fails to save.